src/services/services.ts

```diff
@@ -1503,132 +1503,23 @@ namespace ts {
-                    }
-                    // Not a data URL we can parse, skip it
-                    mapFileName = undefined;
-                }
-            }
-            const possibleMapLocations: string[] = [];
-            if (mapFileName) {
-                possibleMapLocations.push(mapFileName);
-            }
-            possibleMapLocations.push(fileName + ".map");
-            for (const location of possibleMapLocations) {
-                const mapPath = toPath(location, getDirectoryPath(fileName), getCanonicalFileName);
-                if (host.fileExists(mapPath)) {
-                    return convertDocumentToSourceMapper(file, host.readFile(mapPath)!, mapPath); // TODO: GH#18217
-                }
-            }
-            return file.sourceMapper = sourcemaps.identitySourceMapper;
-        }
-
-        function makeGetTargetOfMappedPosition<TIn>(
-            extract: (original: TIn) => sourcemaps.SourceMappableLocation,
-            create: (result: sourcemaps.SourceMappableLocation, unmapped: TIn, original: TIn) => TIn
-        ) {
-            return getTargetOfMappedPosition;
-            function getTargetOfMappedPosition(input: TIn, original = input): TIn {
-                const info = extract(input);
-                if (endsWith(info.fileName, Extension.Dts)) {
-                    let file: SourceFileLike | undefined = program.getSourceFile(info.fileName);
-                    if (!file) {
-                        const path = toPath(info.fileName, currentDirectory, getCanonicalFileName);
-                        file = sourcemappedFileCache.get(path);
-                    }
-                    if (!file) {
-                        return input;
-                    }
-                    const mapper = getSourceMapper(info.fileName, file);
-                    const newLoc = mapper.getOriginalPosition(info);
-                    if (newLoc === info) return input;
-                    return getTargetOfMappedPosition(create(newLoc, input, original), original);
-                }
-                return input;
-            }
-        }
-
-        const getTargetOfMappedDeclarationInfo = makeGetTargetOfMappedPosition(
-            (info: DefinitionInfo) => ({ fileName: info.fileName, position: info.textSpan.start }),
-            (newLoc, info, original) => ({
-                containerKind: info.containerKind,
-                containerName: info.containerName,
-                fileName: newLoc.fileName,
-                kind: info.kind,
-                name: info.name,
-                textSpan: {
-                    start: newLoc.position,
-                    length: info.textSpan.length
-                },
-                originalFileName: original.fileName,
-                originalTextSpan: original.textSpan
-            })
-        );
-
         function getTargetOfMappedDeclarationFiles(infos: ReadonlyArray<DefinitionInfo> | undefined): DefinitionInfo[] | undefined {

```

Comment:
This is hard to do because of the call to `sourceMapper.clearCache()` in `services.ts`. @weswigham There's probably a better signal we could use on when to clear the cache?

Reply:
It's not being evicted correctly right now anyway, since it's evicted on program out of date and not file out of date (because services doesn't track up-to-date ness). Now that it's in server, you should actually be able to just evict the files which are out of date, since the cross-project fs watcher/up-to-date checking stuff should be available. Specifically a file in that cache aughta be invalidated when either the associated file or the associated map updates on disk. Also the cache should likely be shared across all projects, if at all possible - I remarked as much with Ryan when I first implemented it. Honestly it makes way more sense in session because of all that.